sdk/metric/instrument_provider.go

```diff
@@ -0,0 +1,154 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build go1.18
+// +build go1.18
+
+package metric // import "go.opentelemetry.io/otel/sdk/metric"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/asyncfloat64"
+	"go.opentelemetry.io/otel/metric/instrument/asyncint64"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/metric/view"
+)
+
+type asyncInt64Provider struct {
+	scope    instrumentation.Scope
+	registry *pipelineRegistry[int64]
+}
+
+var _ asyncint64.InstrumentProvider = asyncInt64Provider{}
+
+// Counter creates an instrument for recording increasing values.
+func (p asyncInt64Provider) Counter(name string, opts ...instrument.Option) (asyncint64.Counter, error) {
+	cfg := instrument.NewConfig(opts...)
+
+	aggs, err := p.registry.createAggregators(view.Instrument{
+		Scope:       p.scope,
+		Name:        name,
+		Description: cfg.Description(),
+		Kind:        view.AsyncCounter,
+	}, cfg.Unit())
+	if len(aggs) == 0 {
+		if err != nil {
+			err = fmt.Errorf("instrument does not match any view")
+		} else {
+			err = fmt.Errorf("instrument does not match any view: %w", err)
+		}
```

Comment:
How come there's an error when the instrument matches no view?

Reply:
You are right we could drop all instances.  

Maybe this should only be added if we have an error, to notify users that an error occurred and the instrument provided isn't in any view because of it.